### PR TITLE
[fix bug 1122519] CSS/SVG line animation does not work in IE (Edge mode) Win 10

### PR DESCRIPTION
**Pull request by alexgibson**
We need to update our UA check for IE Edge, which sadly still can't animate SVG paths using CSS (even though it "supports" all the right bits). :sob: 

Tested with the following UA: `Mozilla/5.0 (Windows NT 6.4; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.143 Safari/537.36 Edge/12.0`

### CODE REVIEW

**Comment by jpetto** [comment]
UA checks are the worst.

Anyway, code changes work just fine. Two thoughts:

- Should we make a simple helper JS lib (`svganimationcheck.js` or something) to include on these two pages to be a little more DRY? We seem to be fond of these line animations at the moment - might be nice for the (potential) next page.

- Testing for `Edge` in the UA string feels scary. I could see that existing in another UA string, like Canary or Safari beta or something. Should we maybe test for the presence of both `Windows` and `Edge`? Wouldn't be a terribly hard regex change, right?

What do you think?

**Comment by alexgibson** [comment]
> Should we make a simple helper JS lib (svganimationcheck.js or something) to include on these two pages to be a little more DRY? We seem to be fond of these line animations at the moment - might be nice for the (potential) next page.

Good call, will do :+1: 

> Testing for Edge in the UA string feels scary. I could see that existing in another UA string, like Canary or Safari beta or something. Should we maybe test for the presence of both Windows and Edge? Wouldn't be a terribly hard regex change, right?

I was going off what [Polymer are doing](https://github.com/jacobrossi/webcomponentsjs/commit/db767e77697e9b096e3d7367871e082d87b6002d). Will look to see if there's anything a little more robust though.

**Comment by alexgibson** [comment]
Updated but marking this as do not merge for now - I may have a feature detect :+1:

**Comment by alexgibson** [comment]
Ok I think I have a JS feature detect (or rather a series of detects) that should cover both animations - I've tested this on as many browsers as I have and it seems to work. Also tested in IE Edge preview and it correctly gives the fallback :+1: 

Have pushed this to demo2 for easier testing:

https://www-demo2.allizom.org/en-US/firefox/desktop/
https://www-demo2.allizom.org/en-US/firefox/hello/

File: media/css/base/svg-animation-check.less [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    overflow: hidden;
+}
+
+#svg-test-circle {
+    stroke-dasharray: 117;
+    stroke-dashoffset: 117;
+    -webkit-animation: svg-dash-test 0.1s linear;
+    -webkit-animation-fill-mode: forwards;
+    animation: svg-dash-test 0.1s linear;
+    animation-fill-mode: forwards;
```
Comment (jpetto) [inline media/css/base/svg-animation-check.less]: Small perf nit - could consolidate and put `forwards` in `animation` declaration above?

File: media/js/base/svg-animation-check.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * Try to animate stroke-dashoffset using CSS and then get the computed value to see if it has changed.
+ */
+Mozilla.SVGAnimCheck.supportsCSSAnimatedPaths = function() {
+    var div = document.createElement('div');
+    var circle;
+    var offset;
+
+    div.id = 'svg-test';
+    div.innerHTML = '<svg role="presentation" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="0 0 998 450" preserveAspectRatio="xMinYMin meet"><path id="svg-test-circle" stroke-linejoin="round" stroke-linecap="round" stroke-width="2" stroke-opacity="1" fill-opacity="0" d="M33.665530413296274,58.589001490321166C43.94406883919239,58.59306994020939,52.274,66.92651000976564,52.274,77.205C52.274,87.486,43.939,95.821,33.658,95.821C23.377000000000002,95.821,15.042000000000002,87.48599999999999,15.042000000000002,77.205C15.041,66.923,23.376,58.589,33.658,58.589" stroke="#5d7489" fill="#5d7489"></path></svg>';
```
Comment (jpetto) [inline media/js/base/svg-animation-check.js]: Small perf question - do we need all the SVG attributes - `linejoin`, `linecap`, `fill`, etc?

File: media/js/base/svg-animation-check.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    document.documentElement.appendChild(div);
```
Comment (jpetto) [inline media/js/base/svg-animation-check.js]: Should remove the `div` after the test completes?

**Comment by jpetto** [comment]
Tested in all the browsers (Fx, Chrome, Safari, IE 8 - 11 & Edge) and it's looking good. This is an awesome feature detect! Should probably release to the general public. :smile:

**Comment by alexgibson** [comment]
Thanks @jpetto! I'll update here soon :smile:

File: media/js/base/svg-animation-check.js (hunk 3 above)
Comment (alexgibson) [inline media/js/base/svg-animation-check.js]: Ah yeah, I meant to do this in the beginning but forgot to tidy up after the test finishes. Good catch

File: media/js/base/svg-animation-check.js (hunk 2 above)
Comment (alexgibson) [inline media/js/base/svg-animation-check.js]: Left over copy-pasta - will remove :+1:

**Comment by alexgibson** [comment]
Thanks @jpetto - updated!

File: media/js/firefox/hello/index.js [hunk 4]
```diff
@@ -2,7 +2,7 @@
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-;(function(w, $, Modernizr) {
+;(function(Mozilla, w, $, Modernizr) {
```
Comment (jpetto) [inline media/js/firefox/hello/index.js]: After a closer look here, I think we can replace `Modernizr` with `Mozilla.SVGAnimCheck.supportsCSSAnimations`. The use of `Modernizr` here was one of caution anyway (only for mobile/small devices that don't support CSS animations) - better to lighten the JS load and use your version?

**Comment by alexgibson** [comment]
Thanks @jpetto - removed the dependency on Modernizr. I left it as a separate commit so easier to see my changes. Will squarsh before we merge :+1:

**Comment by alexgibson** [comment]
Good catch @jpetto, switched to using `Mozilla.SVGAnimCheck.supportsCSSAnimations()` and squarshed :+1:

**Comment by jpetto** [comment]
Looks great! r+ :clap: :clap: :clap:
